### PR TITLE
1191 - prevented going to the submit stage if the other stages are inva…

### PR DESCRIPTION
**Pull request by IonelLupu**
## What was done
prevented going to the submit stage if the other stages are invalid for new deals

#### Before
The terms stage was allowing to proceed if other stages were invalid

### CODE REVIEW

File: src/wizards/tokenSwapDealWizard/wizardManager.ts [hunk 1]
```diff
@@ -188,11 +188,21 @@ export class WizardManager implements IRouteableComponent {
...
   public async next() {
     const result = await this.controller.validate();
+    this.stages[this.activeIndex].valid = result.valid;
     if (!result.valid) {
       this.eventAggregator.publish("handleValidationError", "Unable to proceed, please check the page for validation errors");
       return;
     }
 
+    // Make sure all the stages are valid before continuing
+    if (this.wizardType === WizardType.createOpenProposal || this.wizardType === WizardType.createPartneredDeal) {
```
Comment (hiaux0) [inline src/wizards/tokenSwapDealWizard/wizardManager.ts:198]: @IonelLupu 
Do you perhaps know why this has to be added in Au2, but was not needed in Au1?

Comment (IonelLupu) [reply src/wizards/tokenSwapDealWizard/wizardManager.ts:198]: it was in AU1 [here](https://github.com/PrimeDAO/prime-deals-dapp/blob/development/src/wizards/services/WizardService.ts#L138)

The wizard was built a bit different but the functionality was almost the same

**Review by hiaux0 — APPROVED** [review]
verified bug
verified fix :checkered_flag:

**Comment by IonelLupu** [comment]
@hiaux0  found a bug in this one. I pushed another fix here: https://github.com/PrimeDAO/prime-deals-dapp/pull/1198

**Pull request merged**